python/tk_desktop/desktop_window.py

```diff
@@ -715,27 +728,28 @@ def __launch_app_proxy_for_project(self, project, pipeline_configuration_id=None
             pipeline_configuration = None
             primary_pipeline_configuration = None
             for pc in pipeline_configurations:
+                # If we've stumbled upon the Primary.
                 if pc["code"] == constants.PRIMARY_PIPELINE_CONFIG_NAME:
                     primary_pipeline_configuration = pc
+                    # And there was no pipeline configuration saved from a last run.
                     if pipeline_configuration_id == 0:
+                        # We'll use it and call it a day!
                         pipeline_configuration = pc
 
-                    if pipeline_configuration is not None:
+                    if pipeline_configuration:
                         break
 
+                # If we have a non primary pipeline configuration and it matches the one we are looking for.
                 if pipeline_configuration_id != 0 and pc["id"] == pipeline_configuration_id:
                     pipeline_configuration = pc
+                    # If we haven't found the primary yet, keep going, otherwise we're done.
                     if primary_pipeline_configuration is not None:
                         break
 
+            # If we haven't found what we were searching for...
             if pipeline_configuration is None:
-                if primary_pipeline_configuration is None:
-                    # Show the Setup Project widget
-                    self.setup_project_widget.project = project
-                    self.setup_project_widget.show()
-                    self.project_overlay.hide()
-                    return
-                else:
+                # ... but the primary exists, switch to that.
+                if primary_pipeline_configuration is not None:
```

Comment:
with boostrap there could be as many as 3 primaries now:

![image](https://cloud.githubusercontent.com/assets/337710/22062724/08c02b28-dd73-11e6-89ca-65153d7e430e.png)

Are we just displaying the "most important" of the three? or all of them?

how is this indicated in the UI? Would be great if we add some comments in the code to explain this particular case!